cli/src/cli_util.rs

```diff
@@ -1012,7 +1012,20 @@ impl WorkspaceCommandHelper {
         let immutable = revset_util::parse_immutable_expression(&self.revset_parse_context())?;
         let mut expression = self.attach_revset_evaluator(immutable)?;
         expression.intersect_with(&to_rewrite_revset);
-        if let Some(commit_id) = expression.evaluate_to_commit_ids()?.next() {
+
+        let mut commit_id_iter = match expression.evaluate_to_commit_ids() {
+            Ok(t) => Ok(t),
+            Err(UserRevsetEvaluationError::Resolution(RevsetResolutionError::NoSuchRevision {
+                name,
+                candidates: _,
+            })) => Err(user_error_with_hint(
+                format!(r#"The revision "{name}" used in immutable_heads() does not exist"#),
+                r#"If you configured specific branch names consider wrapping them with "present()" or using the functions "branches()" / "remote_branches()"."#,
+            )),
+            Err(e) => Err(e.into()),
```

Comment:
Can the first case and this case be combined into `other => other?`?

Reply:
I tried that, but since user_error(_with_hint) returns a `CommandError` I sadly have to call `e.into()` explicitly

Reply:
I see. I'm fine with either that or keeping it as is.